src/api/app/models/bs_request.rb

```diff
@@ -714,6 +714,21 @@ def calculate_state_from_reviews
     end
   end
 
+  def find_review_for_opts(opts)
+    reviews.reverse_each do |review|
```

Comment:
Didn't test it and didn't check the docs, but this should work `reviews.reverse.find { |review| review.reviewable_by?(opts) }` and IIRC it returns nil if no match was found

Reply:
argueing in one line about performance and in another suggesting .reverse instead of .reverse_each is somewhat inconsistent :)

But I can do that.